_data-prepper/log-analytics.md

```diff
@@ -0,0 +1,193 @@
+
+Data Prepper 1.2 release and above comes with a [Grok Processor](https://github.com/opensearch-project/data-prepper/blob/main/data-prepper-plugins/grok-processor/README.md). The Grok Processor is an invaluable tool to structure and extract important fields from your logs in order to make them more queryable.
+
+The Grok Processor comes with a wide variety of [default patterns](https://github.com/thekrakken/java-grok/blob/master/src/main/resources/patterns/patterns) that match against common log formats like apcahe logs or syslogs, but can easily accept any custom patterns that cater to your specific log format.
+
+There are a lot of complex Grok features that will not be discussed here, so please read the documentation if you are interested.
+
+# OpenSearch sink
+
+There is a generic sink that writes the data to OpenSearch as the destination. The [OpenSearch sink](https://github.com/opensearch-project/data-prepper/blob/main/data-prepper-plugins/opensearch/README.md) has configuration options related to an OpenSearch cluster like endpoint, SSL/Username, index name, index template, index state management, etc.
+
+# Pipeline configuration
+
+## Example pipeline.yaml with SSL and basic authentication enabled
+
+Example `pipeline.yaml` with SSL and basic authentication enabled for the `http-source`:
+
+```yaml
+log-pipeline:
+  source:
+    http:
+      ssl: true
+      ssl_certificate_file: "/full/path/to/certfile.crt"
+      ssl_key_file: "/full/path/to/keyfile.key"
+      # The default port that will listen for incoming logs
+      port: 2021
+      authentication:
+        http_basic:
+          username: "myuser"
+          password: "mys3cret"
+  processor:
+    - grok:
+        match:
+          # This will match logs with a "log" key against the COMMONAPACHELOG pattern (ex: { "log": "actual apache log..." } )
+          # You should change this to match what your logs look like. See the grok documenation to get started.
+          log: [ "%{COMMONAPACHELOG}" ]
+  sink:
+    - opensearch:
+        hosts: [ "https://localhost:9200" ]
+        # Change to your credentials
+        username: "admin"
+        password: "admin"
+        # Add a certificate file if you are accessing an OpenSearch cluster with a self-signed certificate  
+        #cert: /path/to/cert
+        # If you are connecting to an Amazon OpenSearch Service domain without
+        # Fine-Grained Access Control, enable these settings. Comment out the
+        # username and password above.
+        #aws_sigv4: true
+        #aws_region: us-east-1
+        # Since we are grok matching for apache logs, it makes sense to send them to an OpenSearch index named apache_logs.
+        # You should change this to correspond with how your OpenSearch indices are set up.
+        index: apache_logs
+```
+
+## Example pipline.yaml without SSL and basic authentication enabled
+
+Example `pipeline.yaml` without SSL and basic authentication enabled for the `http-source`:
+
+```yaml
```

Comment:
This example should end up replacing the following example from OpenSearch.org.

https://opensearch.org/docs/latest/data-prepper/pipelines/#log-ingestion-pipeline

In the current OpenSearch.org example, there are no comments with actions to do. What practice should we take going forward? Should the documentation have clean samples (no extraneous comments) and describe what should change in the subsequent paragraphs? Or should the examples have these commented lines?

@Naarcha-AWS , @carolxob , Thoughts?

Reply:
@dlvenable: I'm okay with code comments within the code examples as long as we've edited for grammar, especially since it removes the need for the developer to refer back documentation should they copy the code.